lib/puppet/indirector/catalog/compiler.rb

```diff
@@ -144,13 +144,13 @@ def inline_metadata(catalog, checksum_type)
 
   # Compile the actual catalog.
   def compile(node, options)
-    if node.environment.static_catalogs? && options[:static_catalog]
+    if node.environment.static_catalogs? && options[:static_catalog] && options[:code_id]
```

Comment:
Other parts of the code check for `node.environment` is not nil, so this access may not be safe. The nil check was added a long time ago, and may not be necessary (outside of rspec tests). But seems safer to do the check now.